doc/mpich/TUNING_PARAMETERS

```diff
@@ -2,3 +2,32 @@ This document discusses some of the new features added to the MPICH-based MPI im
+
+When multiple VCIs are used, MPICH tries to make per-vci progress to avoid lock contention inside the progress engine. However, it also progresses all VCIs, i.e., make global progress, once in a while to avoid potential deadlocks. Such global progress is turned on by default but can be turned off by setting `MPIR_CVAR_CH4_GLOBAL_PROGRESS=0`, which can improve performance under the use of multiple VCIs. However, caution should be taken as it can result in a deadlock.
+
+**3.2 Communicator Info Hints**
+
+When selecting the fine-grain modes (`direct` or `lockless`), messages sent using different communicators can be sent/received using different VCIs. However, messages using the same communicator may have to be sent/received through the same VCI, to guarantee the ordering semantics imposed by the MPI standard. However, messages sent through the same communicator can be sent/received through different VCIs if the MPI runtime knows that wildcards will not be used on that communicator. Thus, MPICH implements the MPI 4.0 per-communicator info hints and recommend users to use them to  indicate the MPI runtime when wildcards are not used, specially when using `direct` or `lockless` threading modes with multiple VCIs if the threads use the same communicator.
+Here are the standard hints from the MPI 4.0 standard that are supported (we recommend using both info hints if the application is not using wildcards):
+
+* `mpi_assert_no_any_source`: Set this info hint if the application does not use the wildcard MPI_ANY_SOURCE on the given communicator.
+* `mpi_assert_no_any_tag`: Set this hint if the application does not use MPI_ANY_TAG on the given communciator.
+
+MPICH also supports a non-standard per-communicator info hint that provides the user with more control of a communicator-VCI mapping:
+* `vci`: Specific sender and receiver VCI can be selected for each communicator using this hint. Applications can utilize this feature to control how messages are spread across the VCIs.
+For instance, programmers can write the applications so that pairs of threads communicate by using different communicator, where each communicator can be mapped to a different VCI through the `vci` info hint. Valid values for this info hints are 0 to (total vcis - 1). This hint is also utilized by MPICH to select a progress thread to process reduction collective operation.
```

Comment:
Could you wrap the text to width of less than 80 chars?

Reply:
Done